e2e/runner/emissions.go

```diff
@@ -0,0 +1,90 @@
+package runner
+
+import (
+	"fmt"
+
+	"cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+
+	"github.com/zeta-chain/node/cmd/zetacored/config"
+	"github.com/zeta-chain/node/e2e/txserver"
+	e2eutils "github.com/zeta-chain/node/e2e/utils"
+	emissionstypes "github.com/zeta-chain/node/x/emissions/types"
+	observertypes "github.com/zeta-chain/node/x/observer/types"
+)
+
+// FundEmissionsPool funds the emissions pool on ZetaChain with the same value as used originally on mainnet (20M ZETA)
+func (r *E2ERunner) FundEmissionsPool() error {
+	r.Logger.Print("⚙️ funding the emissions pool on ZetaChain with 20M ZETA (%s)", txserver.EmissionsPoolAddress)
+
+	return r.ZetaTxServer.FundEmissionsPool(e2eutils.OperationalPolicyName, EmissionsPoolFunding)
+}
+
+// WithdrawEmissions withdraws emissions from the emission pool on ZetaChain for all observers
+// This functions uses the UserEmissionsWithdrawName to create the withdraw tx.
+// UserEmissionsWithdraw can sign the authz transactions because the necessary permissions are granted in the genesis file
+func (r *E2ERunner) WithdrawEmissions() error {
+	observerSet, err := r.ObserverClient.ObserverSet(r.Ctx, &observertypes.QueryObserverSet{})
+	if err != nil {
+		return err
+	}
+
+	for _, observer := range observerSet.Observers {
+		r.Logger.Print("🏃 Withdrawing emissions for observer %s", observer)
+		var (
+			baseDenom            = config.BaseDenom
+			queryObserverBalance = &banktypes.QueryBalanceRequest{
+				Address: observer,
+				Denom:   baseDenom,
+			}
+		)
+
+		balanceBefore, err := r.BankClient.Balance(r.Ctx, queryObserverBalance)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get balance for observer before withdrawing emissions %s", observer)
+		}
+
+		availableAmount, err := r.EmissionsClient.ShowAvailableEmissions(
+			r.Ctx,
+			&emissionstypes.QueryShowAvailableEmissionsRequest{
+				Address: observer,
+			},
+		)
+		if err != nil {
+			return fmt.Errorf("failed to get available emissions for observer %s: %w", observer, err)
+		}
+
+		availableCoin, err := sdk.ParseCoinNormalized(availableAmount.Amount)
+		if err != nil {
+			return fmt.Errorf("failed to parse coin amount: %w", err)
+		}
+
+		if availableCoin.Amount.IsZero() {
+			r.Logger.Print("no emissions to withdraw for observer %s", observer)
+			continue
+		}
```

Comment:
Does `availableCoin` have an `IsValid` function if I recall correctly? That would be a more complete check instead of `IsZero`.

Reply:
The `IsValid` check returns true for zero amount, as 0 is a valida balance, but we don't want to process that in this test.
It does add a check for validate denom , but  `ParseCoinNormalized` already does that